device/alpaka/src/clusterization/measurement_sorting_algorithm.cpp

```diff
@@ -0,0 +1,46 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Library include(s).
+#include "traccc/alpaka/clusterization/measurement_sorting_algorithm.hpp"
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+#include <thrust/sort.h>
+#endif
+
+namespace traccc::alpaka {
+
+measurement_sorting_algorithm::measurement_sorting_algorithm(vecmem::copy& copy)
+    : m_copy{copy} {}
+
+measurement_sorting_algorithm::output_type
+measurement_sorting_algorithm::operator()(
+    const measurement_collection_types::view& measurements_view) const {
+
+    // Get the number of measurements. This is necessary because the input
+    // container may not be fixed sized. And we can't give invalid pointers /
+    // iterators to Thrust.
+    const measurement_collection_types::view::size_type n_measurements =
+        m_copy.get().get_size(measurements_view);
+
+    // Sort the measurements in place
+    // TODO: This obviously won't work for HIP, but we don't have a HIP
+    // equivalent of Thrust yet.
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    thrust::sort(thrust::device, measurements_view.ptr(),
+                 measurements_view.ptr() + n_measurements,
+                 measurement_sort_comp());
+#elif !defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    std::sort(measurements_view.ptr(), measurements_view.ptr() + n_measurements,
+              measurement_sort_comp());
```

Comment:
Doesn't this assume that the code is host-accessible? What we could do is use #632 to implement a _very inefficient_ but functional sorting algorithm. We can do that later, though.

Reply:
It does, we do in general need a better way of dealing with this.

For CUDA, we should be able to use thrust, HIP -> rocThrust....outside of that, right now it should be a CPU and therefore be host accessible.

The test and validation with rocThrust is still ongoing and is likely going to need some extra work, though ideally that could exist outside of the scope of this PR, as its probably going to need a decent number of follow up PRs (mostly due to the pervasieness of thrust across traccc / detray etc making a "simple swap" less simple).